nav2_controller/src/nav2_controller.cpp

```diff
@@ -449,4 +459,16 @@ bool ControllerServer::getRobotPose(geometry_msgs::msg::PoseStamped & pose)
   return true;
 }
 
+void ControllerServer::speedLimitCallback(const nav2_msgs::msg::SpeedLimit::SharedPtr msg)
+{
+  ControllerMap::iterator it;
+  for (it = controllers_.begin(); it != controllers_.end(); ++it) {
+    if (!msg->percentage) {
+      RCLCPP_ERROR(get_logger(), "Speed limit in absolute values is not implemented yet");
+      throw std::runtime_error("Speed limit in absolute values is not implemented yet");
```

Comment:
Don't throw here, since we're in a callback, it cannot be catch and handled. I think the error is fine and return without doing anything in this case.

Reply:
Fixed starting from https://github.com/ros-planning/navigation2/pull/2074/commits/34e694dbee7af9f805d2a7038b292466e44dc104#diff-d344e376ecbc8e5ecd5045731e13855c6d2a9a955cdaee9b31dddbbee7681466L132
and in unit test: https://github.com/ros-planning/navigation2/pull/2074/commits/34e694dbee7af9f805d2a7038b292466e44dc104#diff-b4571d8a4a9cb6222160b2ef3d874689b3eaf5e86a21928869f3b02a90c6b44fL641 and https://github.com/ros-planning/navigation2/pull/2074/commits/34e694dbee7af9f805d2a7038b292466e44dc104#diff-b4571d8a4a9cb6222160b2ef3d874689b3eaf5e86a21928869f3b02a90c6b44fL707

Reply:
See above comment, same with the odd branching issue

Reply:
Should be removed in https://github.com/ros-planning/navigation2/pull/2074/commits/a18462c5fd0f3c608f18a6cdb204adbfb705b6bf#diff-7ce5c619d2d43062d3d66eff15e01887a0343a5865dfff941c579a6a0696f791L468